src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/JavadocExtension.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package org.quiltmc.community.modes.quilt.extensions
+
+import com.kotlindiscord.kord.extensions.commands.application.slash.publicSubCommand
+import com.kotlindiscord.kord.extensions.extensions.Extension
+import com.kotlindiscord.kord.extensions.extensions.publicSlashCommand
+import com.kotlindiscord.kord.extensions.types.respond
+import dev.kord.rest.builder.message.create.embed
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import org.quiltmc.community.GUILDS
+import org.simpleframework.xml.Element
+import org.simpleframework.xml.ElementList
+import org.simpleframework.xml.Root
+import org.simpleframework.xml.Serializer
+import org.simpleframework.xml.core.Persister
+
+@Suppress("UnusedPrivateMember")
+class JavadocExtension : Extension() {
+	override val name: String = "javadoc"
+
+	private val commonProjects = listOf("qsl", "quilt-mappings", "quilt-loader")
+
+	private val client = HttpClient {
+		expectSuccess = false
+	}
+
+	private val xmlSerializer: Serializer = Persister()
+
+	override suspend fun setup() {
+		GUILDS.forEach { guildId ->
+			publicSlashCommand {
+				name = "javadoc"
+				description = "Get Javadocs for common Quilt Projects"
+
+				allowInDms = true
+
+				guild(guildId)
+
+				action {
```

Comment:
This action block can never run. Discord does not support executable root commands when there are groups or subcommands. I'm pretty sure KordEx throws when you do this - you do need to actually test your code.

Reply:
That's annoying but ok. Ill fix this today